coap/adapter.go

```diff
@@ -0,0 +1,60 @@
+package adapter
+
+import (
+	"encoding/json"
+	"log"
+	"net"
+
+	"github.com/dustin/go-coap"
+	"github.com/mainflux/mainflux/writer"
+	broker "github.com/nats-io/go-nats"
+	"go.uber.org/zap"
+)
+
+type Observer struct {
+	conn    *net.UDPConn
+	addr    *net.UDPAddr
+	message *coap.Message
+}
+
+type CoAPAdapter struct {
+	obsMap map[string][]Observer
+	logger *zap.Logger
+	repo   writer.MessageRepository
+}
+
+// NewCoAPAdapter creates new CoAP adapter struct
+func NewCoAPAdapter(logger *zap.Logger, repo writer.MessageRepository) *CoAPAdapter {
+	ca := &CoAPAdapter{
+		logger: logger,
+		repo:   repo,
+	}
+
+	ca.obsMap = make(map[string][]Observer)
+
+	return ca
+}
+
+// Serve function starts CoAP server
+func (ca *CoAPAdapter) Serve(addr string) error {
+	ca.logger.Info("Starting CoAP server", zap.String("address", addr))
+	return coap.ListenAndServe("udp", addr, ca.COAPServer())
+}
+
+// MsgHandler functions is a handler for messages recieved via NATS
+func (ca *CoAPAdapter) MsgHandler(nm *broker.Msg) {
```

Comment:
Either use it as anonymous, or name it properly.

Reply:
It is used twice: https://github.com/mainflux/mainflux/blob/coap/cmd/coap/main.go#L39-L40, I would like to avoid inlining two times.